pkg/daemon/ceph/client/filesystem.go

```diff
@@ -191,6 +191,37 @@ func SetNumMDSRanks(context *clusterd.Context, clusterInfo *ClusterInfo, fsName
 	return nil
 }
 
+// FailAllStandbyReplayMDS: fail all mds in up:standby-replay state
+func FailAllStandbyReplayMDS(context *clusterd.Context, clusterInfo *ClusterInfo, fsName string) error {
+	fs, err := GetFilesystem(context, clusterInfo, fsName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get ceph fs dump")
+	}
+	for _, info := range fs.MDSMap.Info {
+		if info.State == "up:standby-replay" {
+			if err := FailMDS(context, clusterInfo, info.GID); err != nil {
+				return errors.Wrapf(err, "failed to fail mds in up:standby-replay state")
+			}
+		}
+	}
+	return nil
+}
+
+// GetMdsIdByRank get mds ID from the given rank
+func GetMdsIdByRank(context *clusterd.Context, clusterInfo *ClusterInfo, fsName string, rank int32) (string, error) {
+
+	fs, err := GetFilesystem(context, clusterInfo, fsName)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get ceph fs dump")
+	}
+	gid := fs.MDSMap.Up[fmt.Sprintf("mds_%d", rank)]
```

Comment:
Need to handle the case where `mds_<rank>` is not in the MDS map. i.e., `ok == false`